xwiki-commons-core/xwiki-commons-configuration/xwiki-commons-configuration-default/src/main/java/org/xwiki/configuration/internal/SystemEnvConfigurationSource.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.configuration.internal;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import jakarta.inject.Named;
+import jakarta.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+
+/**
+ * Environment variables based configuration source.
+ *
+ * @version $Id$
+ * @since 17.4.0RC1
+ */
+@Component
+@Singleton
+@Named(SystemEnvConfigurationSource.HINT)
+public class SystemEnvConfigurationSource extends AbstractPropertiesConfigurationSource
+{
+    /**
+     * The hint to use to get this configuration source.
+     */
+    public static final String HINT = "envvars";
```

Comment:
sounds like you hesitated a bit on the names, why not `systemenv` like you have `systemproperties`? would be more consistent with the classname no? I noticed you also used the `varsSource` field in SystemConfigurationSource so maybe you wanted to name it `EnvVarConfigurationSource`?